input/script_utils_test.go

```diff
@@ -2034,3 +2035,212 @@ func TestSpecificationKeyDerivation(t *testing.T) {
 			actualRevocationPrivKeyHex)
 	}
 }
+
+// BenchmarkScriptBuilderAlloc benchmarks the script builder's default
+// allocation.
+func BenchmarkScriptBuilderAlloc(t *testing.B) {
+	dummyData := []byte("dummy data")
+	randomPriv, err := btcec.NewPrivateKey()
+	require.NoError(t, err)
+	randomPub := randomPriv.PubKey()
+	randomPubBytes := randomPub.SerializeCompressed()
+
+	t.ReportAllocs()
+	t.ResetTimer()
+
+	for i := 0; i < t.N; i++ {
+		for j := 0; j < 1000; j++ {
+			err := runScriptAllocTest(
+				dummyData, randomPubBytes, randomPub,
+			)
+			if err != nil {
+				t.Fatal(err)
+			}
+		}
+	}
+}
+
+// TestScriptBuilderAlloc makes sure the scripts generated by the utils have the
+// correct length.
+func TestScriptBuilderAlloc(t *testing.T) {
+	t.Parallel()
+
+	dummyData := []byte("dummy data")
+	randomPriv, err := btcec.NewPrivateKey()
+	require.NoError(t, err)
+	randomPub := randomPriv.PubKey()
+	randomPubBytes := randomPub.SerializeCompressed()
+
+	err = runScriptAllocTest(dummyData, randomPubBytes, randomPub)
+	require.NoError(t, err)
+}
+
+// runScriptAllocTest runs the script size test. We don't use the "require"
+// library to assert the length to not influence the benchmark too much.
```

Comment:
I don't understand the second sentence. Errors returned by this function turn into `t.Fatal(err)`. And the function tests for precise lengths.

"to not influence the benchmark too much" what does that mean?

Reply:
Because a call to `require` is a function call, there are additional allocations being made, which influences the overall memory statistics assembled by the benchmark framework. See discussion here: https://github.com/lightningnetwork/lnd/pull/7464#discussion_r1121173590